### PR TITLE
add a new blog

### PR DESCRIPTION
**Pull request by langyuemeng**
@husiyu @YiniXu9506 @csuzhangxc PTAL

### CODE REVIEW

File: dm-source-code-reading-2.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+---
+
+
+本文为 DM 源码阅读系列文章的第二篇，[第一篇](https://www.pingcap.com/blog-cn/dm-source-code-reading-1/)文章简单介绍了 DM 源码阅读的目的和源码阅读的规划、介绍了 DM 的源码结构以及工具链。
+
+从本篇文章开始，我们会正式开始阅读 DM 的源码。本篇文章主要介绍 DM 的整体架构，包括 DM 有哪些组件、各组件分别实现什么功能、组件之间交互的数据模型和 RPC 实现。
+
+## 整体架构
+
+![图 1](media/dm-source-code-reading-2.md/1.png)
```
Comment (YiniXu9506) [inline dm-source-code-reading-2.md]: This is an incorrect image path. Please replace it.

File: dm-source-code-reading-2.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
+---
+title: DM 源码阅读系列文章（二）整体架构介绍
+author: ['张学程']
+date: 2019-03-25
+summary: 本文为 DM 源码阅读系列文章的第二篇，第一篇文章简单介绍了 DM 源码阅读的目的和源码阅读的规划、介绍了 DM 的源码结构以及工具链。
```
Comment (csuzhangxc) [inline dm-source-code-reading-2.md]: how about putting `本篇文章主要介绍 DM 的整体架构，包括 DM 有哪些组件、各组件分别实现什么功能、组件之间交互的数据模型和 RPC 实现。` as the summary?

File: dm-source-code-reading-2.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+tags: ['DM 源码阅读','社区']
+---
+
+
+本文为 DM 源码阅读系列文章的第二篇，[第一篇文章](https://www.pingcap.com/blog-cn/dm-source-code-reading-1/) 简单介绍了 DM 源码阅读的目的和源码阅读的规划、介绍了 DM 的源码结构以及工具链。
```
Comment (csuzhangxc) [inline dm-source-code-reading-2.md]: link to the path in this repo?

File: dm-source-code-reading-2.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+在 [`Worker`](https://github.com/pingcap/dm/blob/3fcf24daa5/dm/worker/worker.go#L43) 这个 struct 中，主要 fields 包括：
+
+*   `subTasks`：维护该 DM-worker 上的所有同步子任务信息。
+
+*   `relayHolder`：对 relay 处理单元相关操作进行简单封装，转发相关操作请求给 relay 处理单元，获取 relay 处理单元的状态信息。
+
+*   `relayPurger`：根据用户配置及相关策略，尝试定期对 relay log 进行 purge 操作。
+
+数据同步子任务管理的代码实现主要在 [`dm/worker/subtask.go`](https://github.com/pingcap/dm/blob/3fcf24daa5/dm/worker/subtask.go) 中， relay 处理单元管理的代码实现主要在 [`dm/worker/relay.go`](https://github.com/pingcap/dm/blob/3fcf24daa5/relay/purger/) 中，对 relay log 进行 purge 操作的代码实现主要在 [`relay/purger`](https://github.com/pingcap/dm/blob/3fcf24daa5/relay/purger/) pkg 中。在本篇文章中，我们暂时只关注 DM 架构相关的实现，上述各功能的具体实现将在后续的相关文章中展开介绍。
```
Comment (csuzhangxc) [inline dm-source-code-reading-2.md]:
```suggestion
数据同步子任务管理的代码实现主要在 [`dm/worker/subtask.go`](https://github.com/pingcap/dm/blob/3fcf24daa5/dm/worker/subtask.go) 中， relay 处理单元管理的代码实现主要在 [`dm/worker/relay.go`](https://github.com/pingcap/dm/blob/3fcf24daa5/dm/worker/relay.go) 中，对 relay log 进行 purge 操作的代码实现主要在 [`relay/purger`](https://github.com/pingcap/dm/blob/3fcf24daa5/relay/purger/) pkg 中。在本篇文章中，我们暂时只关注 DM 架构相关的实现，上述各功能的具体实现将在后续的相关文章中展开介绍。
```

File: dm-source-code-reading-2.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+2.  判断是否需要退出命令行交互环境（exit 命令）或需要进行处理
+
+3.  调用 `ctl.Start` 进行命令分发与处理
+
+dmctl 的具体命令处理实现在 [`dm/ctl`](https://github.com/pingcap/dm/blob/3fcf24daa5/dm/ctl/) pkg 中，入口为 [`dm/ctl/ctl.go`](https://github.com/pingcap/dm/blob/3fcf24daa5/dm/ctl/ctl.go) 中的 [`Start`](https://github.com/pingcap/dm/blob/3fcf24daa5/dm/ctl/ctl.go#L46) 方法，命令的分发与参数解析借助于 [spf13/cobra](https://github.com/spf13/cobra)。命令的具体功能实现在相应的子 pkg 中：
+
+*   `master`：dmctl 与 DM-master 交互的命令，是当前 DM 推荐的命令交互方式。
+
+*   `worker`：dmctl 与 DM-worker 交互的命令，主要用于开发过程中进行 debug，当前并没有实际所有 DM-worker 支持的命令，未来可能废弃。
```
Comment (csuzhangxc) [inline dm-source-code-reading-2.md]:
```suggestion
*   `worker`：dmctl 与 DM-worker 交互的命令，主要用于开发过程中进行 debug，当前并没有实现所有 DM-worker 支持的命令，未来可能废弃。
```

**Comment by husiyu** [comment]
LGTM

**Pull request merged**